sdk/src/client/api/block_builder/input_selection/requirement/mana.rs

```diff
@@ -268,20 +283,32 @@ impl InputSelection {
             required_mana += self.remainder_outputs().map(|o| o.mana()).sum::<u64>() + self.remainders.added_mana;
         }
 
+        Ok((self.total_selected_mana(None)?, required_mana))
+    }
+
+    pub(crate) fn total_selected_mana(&self, include_generated: impl Into<Option<bool>> + Copy) -> Result<u64, Error> {
         let mut selected_mana = 0;
 
         for input in &self.selected_inputs {
-            selected_mana += self.total_mana(input)?;
+            selected_mana += self.total_mana(input, include_generated)?;
         }
-        Ok((selected_mana, required_mana))
+
+        Ok(selected_mana)
     }
 
-    fn total_mana(&self, input: &InputSigningData) -> Result<u64, Error> {
+    fn total_mana(&self, input: &InputSigningData, include_generated: impl Into<Option<bool>>) -> Result<u64, Error> {
```

Comment:
Since this fn is private I'd take a bool directly and do the unwrap/default in `total_selected_mana` instead

Reply:
Otherwise we do this in a loop for nothing

Reply:
fair point

Reply:
actually, the problem is then we have to add the check for the burn at every call point